packages/ra-core/src/controller/list/useListParams.spec.tsx

```diff
@@ -414,21 +441,37 @@ describe('useListParams', () => {
             };
 
             render(
-                <CoreAdminContext
-                    history={history}
-                    dataProvider={testDataProvider()}
+                <TestMemoryRouter
+                    locationCallback={l => {
+                        location = l;
+                    }}
                 >
-                    <Component />
-                    <StoreReader />
-                </CoreAdminContext>
+                    <CoreAdminContext dataProvider={testDataProvider()}>
+                        <Component disableSyncWithLocation />
```

Comment:
I suspect this test was buggy in the first place, as it did not set `disableSyncWithLocation` although it is necessary to disable sync. Hopefully by adding `await screen.findByText('page: 10');` I have fixed it.